### PR TITLE
implementing a global tools-view component + schema.org support

### PR DESCRIPTION
**Pull request by OlegZharkov**
Porting tools/view from https://github.com/usegalaxy-eu/galaxy/pull/17 and https://github.com/usegalaxy-eu/galaxy/pull/18 .

--------

This PR is adding a `/tools/view` and a `/tools/json` endpoint. The tools/view is displaying all tools on one page so users can scroll through it. It is meant for better findability and also to give one page to get an overview of all available tools. The /tools/json view is returning all tools as schemas.org annotation. Hopefully, Google and Co. will index this page and offer tools from various servers to users.

### CODE REVIEW

File: client/galaxy/scripts/components/ToolsView.vue [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+                        <b-link :href="info['url']" target="_blank">
+                            <h4 class="tools-view-name">{{ info["name"] }}</h4>
+                        </b-link>
+                        <b-badge class="tools-view-section">{{ info["section"] }}</b-badge>
+                    </div>
+                    <p class="card-text" v-html="helpSummary(info['help']) || info['description']" />
+                    <p class="card-text">
+                        <b-btn v-b-modal="'modal-' + '-' + index">Info</b-btn>
+                        <b-modal :id="'modal-' + '-' + index" centered :title="info['name']">
+                            <b>{{ info["version"] + " / " + info["id"] }}</b>
```
Comment (guerler) [inline client/galaxy/scripts/components/ToolsView.vue]: Please use info.version instead. Thank you for working on this.

Comment (OlegZharkov) [reply client/galaxy/scripts/components/ToolsView.vue]: Thanks for checking! Will be done

**Comment by guerler** [comment]
@OlegZharkov fyi the SidePanel.vue component has been moved to components/Panels/SidePanel.vue.

File: lib/galaxy/tools/__init__.py [hunk 2]
```diff
@@ -1954,6 +1954,13 @@ def to_dict(self, trans, link_details=False, io_details=False):
         # FIXME: the Tool class should declare directly, instead of ad hoc inspection
         regular_form = tool_class == Tool or isinstance(self, (DatabaseOperationTool, InteractiveTool))
         tool_dict["form_style"] = "regular" if regular_form else "special"
+        if tool_help:
+            # create tool help
+            help_txt = ''
+            if self.help:
+                help_txt = self.help.render(static_path=self.app.url_for('/static'), host_url=self.app.url_for('/', qualified=True))
+                help_txt = unicodify(help_txt, 'utf-8')
```
Comment (nsoranzo) [inline lib/galaxy/tools/__init__.py]:
```suggestion
                help_txt = unicodify(help_txt)
```

File: lib/galaxy/tools/toolbox/base.py [hunk 3]
```diff
@@ -1073,9 +1078,9 @@ def to_dict(self, trans, in_panel=True, **kwds):
...
                 # Only use cache for objects that are Tools.
                 if hasattr(elt, "tool_type"):
-                    rval.append(self.get_tool_to_dict(trans, elt))
+                    rval.append(self.get_tool_to_dict(trans, elt, tool_help=tool_help))
                 else:
-                    kwargs = dict(trans=trans, link_details=True, toolbox=self)
+                    kwargs = dict(trans=trans, link_details=True, tool_help=tool_help, toolbox=self)
                     rval.append(elt.to_dict(**kwargs))
         else:
             filter_method = self._build_filter_method(trans)
```
Comment (nsoranzo) [inline lib/galaxy/tools/toolbox/base.py:1091]: There's another `get_tool_to_dict()` call at line 1091 below where `, tool_help=tool_help` should also probably be added.

Comment (OlegZharkov) [reply lib/galaxy/tools/toolbox/base.py:1091]: thanks! fixed

**Comment by nsoranzo** [comment]
@galaxybot test this

**Comment by nsoranzo** [comment]
@galaxybot test this

File: client/yarn.lock [hunk 4]
```diff
@@ -14605,10 +14668,10 @@ webpack-dev-server@^3.1.14:
     webpack-log "^2.0.0"
     yargs "12.0.2"
 
-webpack-dev-server@^3.10.1:
-  version "3.10.1"
-  resolved "https://registry.yarnpkg.com/webpack-dev-server/-/webpack-dev-server-3.10.1.tgz#1ff3e5cccf8e0897aa3f5909c654e623f69b1c0e"
-  integrity sha512-AGG4+XrrXn4rbZUueyNrQgO4KGnol+0wm3MPdqGLmmA+NofZl3blZQKxZ9BND6RDNuvAK9OMYClhjOSnxpWRoA==
+webpack-dev-server@^3.9.0:
```
Comment (jmchilton) [inline client/yarn.lock]: Is this right?

File: client/package.json [hunk 5]
```diff
@@ -31,6 +31,8 @@
     "glob": "^7.1.4",
     "handsontable": "^2.0.0",
     "imask": "^4.1.5",
+    "imports-loader": "^0.8.0",
```
Comment (dannon) [inline client/package.json]: Not sure what went sideways here, but this should only be in devDependencies.  Same as `webpack-dev-server` below.

File: client/yarn.lock (hunk 4 above)
Comment (dannon) [reply client/yarn.lock]: Might be a bad merge, but it's not correct -- good catch.  The package.json is incorrect as well, with a couple duplicated dependencies from devDependencies.

Comment (OlegZharkov) [reply client/yarn.lock]: @dannon you're right. I will remove webpack-dev-server and imports-loader from package.json:dependencies and force-push.

File: client/package.json (hunk 5 above)
Comment (OlegZharkov) [reply client/package.json]: Thanks! it was most probably a merge problem. it should be better now.

**Comment by nsoranzo** [comment]
@galaxybot test this

**Comment by bgruening** [comment]
Add some description at the top. @guerler @dannon thanks for reviewing!

**Comment by OlegZharkov** [comment]
@bgruening > Add some description at the top. 

Sure, which component? schema.org?

**Comment by bgruening** [comment]
Sorry, I meant, I added :) ... Nothing to do for you. Thanks a lot @OlegZharkov

**Comment by OlegZharkov** [comment]
@bgruening Ah, I see. Thanks for description!

**Comment by bgruening** [comment]
@guerler @dannon anything that we can do here?

**Comment by dannon** [comment]
@OlegZharkov Thanks for the ping, sorry for letting this one sit.  I'll go through this today.

(~got a start on it, need a little more time, will finish in the morning~ getting to it now)

**Review by dannon — APPROVED** [review]
Talked offline with @OlegZharkov and we're going to follow up on the schema.org view separately, going ahead and including it for now.  Since it's a page just for indexing, it seems like we probably want it generated on the serverside and sent, instead of shipping a js client route that then makes the API calls to generate a plain text feed.

Going to merge this when tests are sorted out.

**Comment by bgruening** [comment]
Thanks @dannon and @guerler!

**Comment by OlegZharkov** [comment]
Thanks!

**Pull request merged**